pkgs/applications/window-managers/wayfire/windecor.nix

```diff
@@ -0,0 +1,63 @@
+{ stdenv
+, lib
+, fetchFromGitLab
+, meson
+, ninja
+, pkg-config
+, wayfire
+, wf-config
+, wayland
+, pango
+, eudev
+, libinput
+, libxkbcommon
+, librsvg
+, libGL
+, xcbutilwm
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "windecor";
+  version = "0.8.0";
+
+  src = fetchFromGitLab {
+    owner = "wayfireplugins";
+    repo = "windecor";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-v0kGT+KrtfFJ/hp1Dr8izKVj6UHhuW6udHFjWt1y9TY=";
+  };
+
+  postPatch = ''
+    substituteInPlace meson.build \
+      --replace "wayfire.get_variable( pkgconfig: 'metadatadir' )" "join_paths(get_option('prefix'), 'share/wayfire/metadata')"
```

Comment:
Would be nice to upstream this

Reply:
No，`metadatadir` works fine in other distributions，wayfire plugins should be installed in a same directory，we must use [symlinkJoin](https://github.com/NixOS/nixpkgs/blob/a6b7e8804a3a1e48923bb565044e9bba1dec4eb6/pkgs/applications/window-managers/wayfire/wrapper.nix) in nix